src/overlays/actors/ovl_Bg_Bdan_Objects/z_bg_bdan_objects.h

```diff
@@ -6,12 +6,22 @@
 
 struct BgBdanObjects;
 
+typedef enum {
+    BDAN_OBJECT_TYPE_BIG_OCTO_PLATFORM = 0,
+    BDAN_OBJECT_TYPE_SMALL_AUTO_ELEVATOR = 1,
+    BDAN_OBJECT_TYPE_WATERBOX_HEIGHT_CHANGER = 2,
+    BDAN_OBJECT_TYPE_FALLING_PLATFORM = 3
+} BgBdanObjectType;
+
 typedef void (*BgBdanObjectsActionFunc)(struct BgBdanObjects*, PlayState*);
 
 typedef struct BgBdanObjects {
     /* 0x0000 */ DynaPolyActor dyna;
     /* 0x0164 */ BgBdanObjectsActionFunc actionFunc;
+    union {
     /* 0x0168 */ u8 switchFlag;
+    /* 0x0168 */ u8 camChangeTimer;
+    };
```

Comment:
I'm not opposed to using unions, on the contrary, but we avoided these constructs before so afaik there aren't many examples if any of it in the repo

the number one issue is it's not obvious with an anonymous union that the two fields share the same memory (which harms readability). Naming the union fixes that (e.g. `var` then `this->var.switchFlag`, `this->var.camChangeTimer`) but finding a name for it is *bruh* (and again no example of it afaik)

@fig02 ?

Reply:
if we want to start doing the player approach everywhere, yeah, a named union would be better. I dont think we really ironed out that in discussions though. 
Id be fine with a named enum for all cases like this personally

Reply:
named the union `var`, but going to leave this conversation unresolved in case there are other opinions, since we don't really have a strong consensus